common/source/docs/common-downloading-and-analyzing-data-logs-in-mission-planner.rst

```diff
@@ -38,6 +38,22 @@ Replay Logging
 
 ArduPilot has the ability to log in a fashion that solutions to EKF/AHRS issues can be more easily verified by actually re-playing a log against code changes to see if the solution results in the desired, corrected behavior. This requires that the logs showing the issue to be worked on be made with logging active during disarmed periods (with :ref:`LOG_DISARMED<LOG_DISARMED>` set to a non-zero value, preferably 3) and :ref:`LOG_REPLAY<LOG_REPLAY>` =1 , thereby logging more sensor data than normal.
 
+On-Board DataFlash Logging
+==========================
+
+Some boards do not have SD card interfaces for logging, but rather a limited amount of dataflash, typically 16MB. This saves log files in a manner like a circular buffer. Once the flash is filled, the oldest log file is overwritten with the current logging data. If there is only one file on the flash when space runs out, logging is stopped instead.
+
+A new log file will be started after boot, upon arming, or, immediately if :ref:`LOG_DISARMED<LOG_DISARMED>` is 1.
+
+If :ref:`LOG_FILE_DSRMROT<LOG_FILE_DSRMROT>` is enabled, any disarm will stop logging and a new file started upon the next arm or immediately if :ref:`LOG_DISARMED<LOG_DISARMED>` is 1. Otherwise, logging to the current file will resume on a re-arm. Any reboot stops logging to the current file.
+
+In order to maximize the utility of the limited flash space several things can be done:
+
+- Reduce the things logged using :ref:`LOG_BITMASK<LOG_BITMASK>`.
+- Eliminate logging the EKF3 messages which are voluminous and usually needed only for problem diagnosis using the :ref:`EK3_LOG_LEVEL<EK3_LOG_LEVEL>` parameter.
+- Only log when needed during the flight, ie tuning, gathering data for TECS tuning, etc. using an RC Aux switch set to "164" to start and stop log writes.
+- Download and erase the logs each flight and only log one file for a flight
```

Comment:
I think its worth mentioning that flash logging is often slow, so if you try and log too much you will get gaps in the logs. Boards with the W25N01GV are much better because (a) it's 128Mb and (b) its much faster.

Reply:
done...thanks